opamp/config_manager.go

```diff
@@ -0,0 +1,42 @@
+package opamp
+
+import (
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"go.uber.org/zap"
+)
+
+// agentConfigManager is responsible for managing the agent configuration
+// It is responsible for:
+// 1. Reading the agent configuration from the file
+// 2. Reloading the agent configuration when the file changes
+// 3. Providing the current agent configuration to the Opamp client
+type agentConfigManager struct {
+	agentConfig *remoteControlledConfig
```

Comment:
should this be locked to concurrent changes?

Reply:
okay...you can mark the conv resolved when this is done

Reply:
responded to same question here https://github.com/SigNoz/signoz-otel-collector/pull/56#issuecomment-1364638488